### DISCUSSION TITLE
Huawei P10 VTR-L29 (dual SIM) and Lineage 18.1, 19.1, 20

### DISCUSSION THREAD

**Discussion by michal25** [phhusson/treble_experimentations#2542] [Ideas]
Works!
But you have to solve some basic problems:
1. Unlock bootloader. Solved here:  https://github.com/mashed-potatoes/PotatoNV
2. You have to patch the recovery_ramdisk original partition. Patch with The Magisk https://github.com/topjohnwu/Magisk
3. Take the recovery_ramdisk partition from the latest full upgrade file VTR-L29 9.1.0.297(C432E5R1P9), VTR-L29C432B185
4. You have to upgrade the phone firmware to the latest version. 
You can use this: P10_Victoria-L09B 9.1.0.275(C432E2R1P9T8)_Firmware_EMUI9.1.0_05014JGC_[HardReset.info].7z (This archive contains both firmware for VTR-L09 and VTR-L29)
[https://tfmserver.com/index.php?a=downloads&b=tags&tag=VTR-L09 9.1.0.275(C432E2R1P9T8) Firmware](https://tfmserver.com/index.php?a=downloads&b=tags&tag=VTR-L09%209.1.0.275(C432E2R1P9T8)%20Firmware)
5. Download the lineage .img file https://sourceforge.net/projects/andyyan-gsi/files/. You will need the   -arm64_bgN.img.xz version
6. Reboot the phone to fastboot mode and execute command: fastboot flash system xxxxx-arm64_bgN.img
7. fastboot reboot
8. After reboot the phone go to the erecovery (Volume up to 3 secondes) and do wipe data / factory reset
9. Boot phone in to clean Lineage system and do the first Lineage configuration
10. Reboot phone to fastboot mode
11. Flash the Magisk patched recovery_ramdisk partition: fastboot flash recovery_ramdisk magisk_patched_recovery_ramdisk.img
12. Now you have a full working Huawei P10 with Lineage 18.1, 19.1 or 20.x system, but every time you will have to boot the phone with "Magisk boot cheat" means Volume Up + Power button. When appears the Huawei logo release all buttons. https://topjohnwu.github.io/Magisk/install.html
13. Not necessary - permanent boot to recovery cheat. When the phone boot and the yellow text on black wallpaper appers, use the first option and go to eRecovery (Volume Up for 3 seconds). In eRecovery select Wipe data / Factory reset, confirm and reboot the phone. The phone will reboot to recovery and will boot in to the Magisk root boot system (the wipe will not be called).
14. Go back to point 12 state. The RAM card have not to contain a /dload directory!!!!! Power off the phone. Press Volume Up + Volume Down + Power and release all buttons, when Huawei logo appears. Now you will see aclean EMUI screen which will respont - OS Upgrade not successfull. Go to reboot and the permanent Recovery reboot byte will be cleaned.
15. Relevant issue  https://github.com/phhusson/treble_experimentations/wiki/Huawei-P10-and-P10-Plus

**Comment by michal25** [comment]
Screenshots added.
![Huawei-P10-boot-to-recovery-Magisk-patched-boot](https://user-images.githubusercontent.com/5064337/234036290-25fad303-7063-43ad-9747-a5bcc4f34ab5.jpg)
![Huawei-P10-eRecovery-Factory-reset](https://user-images.githubusercontent.com/5064337/234036401-d917bf4e-443c-43d6-b5ac-e7b297ebc481.jpg)
![Huawei-P10-eRecovery-OS-upgrade](https://user-images.githubusercontent.com/5064337/234036411-40b9e505-b092-42b2-9b82-deb5490ac8cb.jpg)
![Huawei-P10-eRecovery-Wipe-data-Factory-reset-1](https://user-images.githubusercontent.com/5064337/234036419-3b9f26f8-9fc4-426f-96b9-7cfb62e06648.jpg)
![Huawei-P10-normal-boot](https://user-images.githubusercontent.com/5064337/234036434-4b83e7c9-e608-41ca-a151-4f7a0405508c.jpg)

**Reply by JonnyTech** [reply]
Or simply just don't use magisk...

**Reply by StancuFlorin** [reply]
Any clue on how to run "stop aptouch" without magisk? I tried this [1] but didn't worked - I guess, "aptouch" service is started after my custom service.

[1] https://ch1p.io/lineageos-run-shell-script-at-boot-as-root/

LE: I was able to run my custom script when the phone starts without magisk
https://notes.stancu.me/2024/01/02/how-to-install-lineage-os-19-on-huawei-p10.html

**Comment by michal25** [comment]
You will need The Magisk, because AdAway is not working withou the Magisk systemless host module.
https://github.com/AdAway/AdAway

**Reply by JonnyTech** [reply]
AdAway can use [root with phh's Superuser](https://f-droid.org/en/packages/me.phh.superuser/) too, I use that on my devices, no need for magisk.
